Winleafs.Models/Models/Layouts/CustomEffect.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+
+namespace Winleafs.Models.Models.Layouts
+{
+    /// <summary>
+    /// A user created Custom Effect which specifies a static pattern or animation
+    /// </summary>
+    public class CustomEffect
+    {
+        /// <summary>
+        /// True if this is just one Frame to set each panel's color
+        /// </summary>
+        public bool IsStatic { get; set; }
+
+        /// <summary>
+        /// True if the animation is played repeatedly
+        /// </summary>
+        public bool IsLoop { get; set; }
+
+        /// <summary>
+        /// A series of frames each specifying the color for each panel
+        /// </summary>
+        public IList<Frame> Frames { get; set; }
```

Comment:
So IMHO IList`<T>` is preferable to `List<T>` so we're not forcing the specific implementation of `List`. `IEnumerable` may be okay if I'm not using any `IList` methods. Sometimes you can end up with LINQ having to provide functionality that is already there, an example is `Count`. `IList` provides a property for `Count`, `IEnumerable` doesn't, so if you need to count the items you have to get LINQ to iterate over the `IEnumerable` for you using `Count()` rather than just using the `Count` property that would be there if it were an `IList`. 

The same is true for `Any()`, rather than `Count > 0`. This isn't as bad as `Count()` as `Any()` just needs to read one item from the `IEnumerable`, but it is still less efficient than the `Count` property.

I'll see what I break if I change to `IEnumerable` and get back to you.

Reply:
There is a lot of use of indexing, so it has to be `List` or `IList`. Personally I prefer `IList` as this means if we ever wanted to swap the implementation from 'List' to a similar collection that still implements 'IList' we could with potentially breaking code.